exercises/react/example.cr

```diff
@@ -0,0 +1,69 @@
+module React
+  alias CellValue = Int32
+
+  private class Cell
+    @dependents = [] of ComputeCell
+
+    def initialize(initial : CellValue)
+      @value = initial
+    end
+
+    getter :value
+
+    protected getter :dependents
+  end
+
+  class InputCell < Cell
+    def value=(new_value : CellValue)
+      @value = new_value
+      @dependents.each(&.propagate)
+      @dependents.each(&.fire_callbacks)
+    end
+  end
+
+  class ComputeCell < Cell
+    alias CallbackID = Int32
+
+    @callbacks_given = 0
+    @callbacks = {} of CallbackID => CellValue ->
+    @last_callback_value : CellValue
+
+    def initialize(input : Cell, &compute : CellValue -> CellValue)
+      initialize(->{ compute.call(input.value) }, input)
+    end
+
+    def initialize(input1 : Cell, input2 : Cell, &compute : (CellValue, CellValue) -> CellValue)
```

Comment:
https://github.com/exercism/xcrystal/pull/44#discussion_r78288531 applies (I regret having to do a different overload per arity, but it seems unavoidable if we wish to have the compiler check that the arities match up)